android/app/src/main/java/covidsafepaths/bt/exposurenotifications/DebugMenuModule.java

```diff
@@ -49,50 +46,65 @@ String getName() {
     }
 
     @ReactMethod
-    public void submitExposureKeys(Promise promise) {
-        Activity activity = getCurrentActivity();
-        if (activity instanceof MainActivity) {
-            ((MainActivity) activity).share(promise);
-        }
+    public void fetchDebugLog(Promise promise) {
+        promise.resolve("AppId: " + BuildConfig.ANDROID_APPLICATION_ID +
+                "\nVersion: " + BuildConfig.VERSION_NAME +
+                "\nVersionName: " + BuildConfig.VERSION_CODE +
+                "\nLastProcessedFileName: " + RealmSecureStorageBte.INSTANCE.getLastProcessedKeyZipFileName() +
+                "\nLastDetectionProcessDate: " + prefs.getLastDetectionProcessDate());
+    }
+
+    @ReactMethod
+    public void forceAppCrash(Promise promise) throws Exception {
+        throw new Exception("Force crash");
+    }
+
+    @ReactMethod
+    public void fetchDiagnosisKeys(Promise promise) {
+        ExposureNotificationClientWrapper client = ExposureNotificationClientWrapper.get(getReactApplicationContext());
+        client.getTemporaryExposureKeyHistory()
+                .addOnSuccessListener(keys -> {
+                    List<RNDiagnosisKey> diagnosisKeys = new ArrayList<>();
+                    for (TemporaryExposureKey key : keys) {
+                        RNDiagnosisKey diagnosisKey = new RNDiagnosisKey(
+                                key.getRollingStartIntervalNumber()
+                        );
+                        diagnosisKeys.add(diagnosisKey);
+                    }
+
+                    promise.resolve(Util.convertListToWritableArray(diagnosisKeys));
+                })
+                .addOnFailureListener(exception -> {
+                    if (!(exception instanceof ApiException)) {
+                        promise.reject(exception);
+                        return;
+                    }
+
+                    ApiException apiException = (ApiException) exception;
+                    if (apiException.getStatusCode() == ExposureNotificationStatusCodes.RESOLUTION_REQUIRED) {
+                        // TODO Refactor MainActivity/ExposureNotificationClientWrapper to call this method after permission is accepted
```

Comment:
Did you mean to address this TODO in this PR? All good if not, looks like something that can be done later.

Reply:
I'll address this in a new PR, it requires some changes to the existing functionality that I prefer to test separately